src/components/AddTodoForm/AddTodoForm.tsx

```
@@ -0,0 +1,109 @@
+import { useState } from 'react';
+import { Nullable } from '../..//components/domain/Nullable';
+import usersFromServer from '../../api/users';
+import { Todo } from '../domain/Todo';
+
+type AddTodoFormProps = {
+  onSubmit: (todo: Omit<Todo, 'id'>) => void;
+};
+
+export const AddTodoForm = ({ onSubmit }: AddTodoFormProps) => {
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState<Nullable<string>>(null);
+
+  const [userId, setUserId] = useState(0);
+  const [userIdError, setUserIdError] = useState<Nullable<string>>(null);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newTitle = event.target.value;
+
+    setTitle(newTitle);
+
+    if (titleError && newTitle.trim().length > 0) {
```

Current logic clears the title error only when `newTitle.trim().length > 0`: `if (titleError && newTitle.trim().length > 0) { setTitleError(null); }`. The checklist requires: "Hide an error message immediately after any change in the corresponding field." Clear the error on any change (when `titleError` exists) rather than waiting for the value to become non-empty. For example, `if (titleError) setTitleError(null);` inside this handler.